lib/bitcrowd_ecto/repo.ex

```diff
@@ -153,7 +153,7 @@ defmodule BitcrowdEcto.Repo do
     |> where([], ^Enum.to_list(clauses))
     |> maybe_apply_lock(opts)
     |> maybe_preload(opts)
-    |> repo.one()
+    |> repo.one(opts)
```

Comment:
I'm wondering whether we should make sure we only pass Ecto's options here, i.e. either by deleting our custom options or by `Keyword.take(opts, [:prefix, :timeout, ...])` taking Ecto's.

Either way, the `fetch_option` type above must be updated (otherwise dialyzer errors), i.e. by adding Ecto's options, or by making a new type.

```
@type ecto_options :: keyword
```

Keep in mind that a new 

```
@spec fetch(module, any, [fetch_option()] | ecto_options()) :: fetch_result
```

is much weaker than before, as dialyzer can't catch you anymore if you mistype our own options. So adding Ecto's options explicitly is better for dialyzer, though worse for maintainability. Not sure on this one..

Plus, please update the docs on `fetch_by` (& friends, they all reference each other somehow IIRC) to mention the fact that you can now pass `Ecto.Repo.one/2` options (mention `prefix` explicitly and the ["Shared Options"](https://hexdocs.pm/ecto/Ecto.Repo.html#module-shared-options) perhaps)

Reply:
I'm fine being explicit with the ecto options. I don't think they will change much.